sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ImpulseTest.java

```diff
@@ -0,0 +1,111 @@
+package org.apache.beam.sdk.transforms;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.apache.beam.runners.core.construction.JavaReadViaImpulse;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.testing.UsesImpulse;
+import org.apache.beam.sdk.testing.ValidatesRunner;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ImpulseTest {
+  @Rule
+  public transient TestPipeline p = TestPipeline.create();
+
+  @Test
+  @Category({ValidatesRunner.class, UsesImpulse.class})
+  public void testImpulseRead() {
+    PCollection<Integer> result = p.apply(JavaReadViaImpulse.bounded(Source.of(1, 2, 3)));
```

Comment:
why not just a `ParDo`?

Reply:
Will pipeline construction _always_ insert an Impulse in that case? Or do I need to insert one myself? I want to ensure that runners can actually handle `Impulse` proper nodes.

Reply:
What I mean is just insert an `Impulse` then a `ParDo` after it, instead of wrapping a `Source`

Reply:
OK, I've added another test with a direct `Impulse` -> `ParDo`. Let me know if this looks good and I'll delete the source-based test.

Reply:
This seeks to work fine and tests what we want to test, so I've gone ahead and removed the source-based test.